code/modules/events/spacevine.dm

```diff
@@ -541,8 +541,20 @@
 		buckle_mob(V, 1)
 
 /obj/structure/spacevine/proc/spread()
-	var/direction = pick(GLOB.cardinals)
-	var/turf/stepturf = get_step(src,direction)
+	var/turf/startturf = get_turf(src)
+	var/direction = pick(GLOB.cardinals_multiz)
+	var/turf/stepturf = get_step_multiz(src,direction)
+	if(direction == UP || direction == DOWN)
+		var/ladder = FALSE
+		for(var/obj/structure/ladder/L in startturf.contents)
+			if(!L.up && direction == UP)
+				continue
+			else if(!L.down && direction == DOWN)
+				continue
+			ladder = TRUE
+		if(!stepturf?.zPassIn(src, direction, startturf) && !startturf.zPassOut(src, direction, stepturf) && !ladder) //We can't zmove, choose another direction
```

Comment:
Should probably fail if either doesn't allow passage, not neccessarily both?
```suggestion
		if((!stepturf?.zPassIn(src, direction, startturf) || !startturf.zPassOut(src, direction, stepturf)) && !ladder) //We can't zmove, choose another direction
```

Reply:
No, because this is direction-dependent. You want to check both if the turf you are entering allows it from your direction, and if the turf you are exiting also allows you to leave in that direction

Reply:
Of course, for turfs like floors they always let you leave and enter from above them, but imagine a vine trying to travel down openspace into a wall below

Reply:
But.. this is the check that if true *prevents* movement is it not?

So right now if you can move out of the current tile, but cannot enter the one you want to go to... the proc is false and allows movement, aka the exact case you describe here, right?
Which means you want to intercept if *either* blocking case if present, and block movement if it is except if a ladder is there, no?

Reply:
Apparently bee did not realize this in my fix and just merged it in 3 days >.>